Add Dockerfile, build pipeline, push to DockerHub

GitBytes opened the pull request:
### Description

Adds a Dockerfile and build pipeline to support container deployment.

### Issues Resolved

Fixes #27 
References https://github.com/pacifica/pacifica/issues/22

### Check List

- [ ] All tests pass. See <https://github.com/pacifica/pacifica-docs/blob/master/development/TESTING.md>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

GitBytes committed:
add basic dockerfile to start building containerized service

GitBytes committed:
adding multistage prod build

GitBytes committed:
adding some readability and clarity for reading travis tests

GitBytes committed:
had to move these to the main deps for now

eslint deps need further evaluation so they aren't necessarily used for prod, maybe

GitBytes commented:
@dmlb2000 I think this is ready for review now. Thanks.

dmlb2000 commented on package.json:
```diff
@@ -12,7 +12,16 @@
     "react-dom": "^16.8.6",
     "react-scripts": "3.0.0",
-    "react-table": "^6.10.0"
+    "react-table": "^6.10.0",
```
All the eslint stuff shouldn't be in the dependencies, it's a dev dependency. Why were these moved here?

GitBytes replied:
Production build was failing otherwise. Maybe I was mistaken? @dmlb2000

dmlb2000 replied:
So as far as I can tell from a bit of searching...

 * https://github.com/facebook/create-react-app/issues/1764
 * https://github.com/facebook/create-react-app/issues/7078
 * https://stackoverflow.com/questions/54677395/error-failed-to-load-plugin-import-cannot-find-module-eslint-plugin-import-w
 * https://create-react-app.dev/docs/setting-up-your-editor#formatting-code-automatically
 * https://create-react-app.dev/docs/production-build

I've come to some conclusions:

  * Seems like the react community doesn't give a damn about trying to keep development dependencies away from production dependencies.
  * The build (`npm run build`) should produce a compiled build for production use.
  * The production build should compile out any code that isn't required directly by the App.
  * The dependencies should be viewed as a single blob just install them all and let the compiler figure it out (hopefully it will).

Makes sense?

GitBytes replied:
That makes sense. I'll make sure there is no split between dev and prod deps then? I can make that change and submit that for approval as well.

dmlb2000 replied:
That's the thought, just do an `npm install` and don't worry about getting a thousands of packages that are only useful when developing the application... *sigh*

dmlb2000 replied:
Should be resolved with my changes.

dmlb2000 reviewed (changes requested):
I would suggest moving the eslint dependencies back to devDependencies then it should be good to go.

dmlb2000 committed:
Revert package move

Signed-off-by: David Brown <dmlb2000@gmail.com>

dmlb2000 committed:
Don't mess with production vs development dependencies

Signed-off-by: David Brown <dmlb2000@gmail.com>

dmlb2000 committed:
Add dockerignore file

Signed-off-by: David Brown <dmlb2000@gmail.com>

dmlb2000 committed:
Copy lock file too

This ensures we are developing on and testing on same things we are
building production builds on.

Signed-off-by: David Brown <dmlb2000@gmail.com>

dmlb2000 reviewed (approved):
I fixed the build the way I like it, should be fine now.

GitBytes commented on Dockerfile:
```diff
@@ -0,0 +1,32 @@
+# ---- Base Node ----
+FROM node:10-stretch AS base
```
@dmlb2000 After doing some additional testing, we should ok to move to alpine based images as it will significantly reduce image size.
